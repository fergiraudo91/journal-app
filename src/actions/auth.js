import Swal from 'sweetalert2';
import { types } from "../types/types";
import { firebase, googleAddProvider } from "../firebase/firebase-config";
import { finishLoading, startLoading } from "./ui";

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch(finishLoading());
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
        dispatch(finishLoading());
      });
  };
};

export const startRegisterEmailPasswordName = (name, email, password) => {
  return (dispatch) => {
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const googleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAddProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      });
  };
};

export const login = (uid, displayName) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName,
    },
  };
};
