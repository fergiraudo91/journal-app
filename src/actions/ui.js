import { types } from "../types/types";
import { firebase } from '../firebase/firebase-config';
import { notesLogout } from "./notes";

export const setError = (err) => ({
    type: types.uiSetErrors,
    payload: err
});


export const removeError = () => ({
    type: types.uiRemoveErrors
});

export const startLoading = () => ({
    type: types.uiStartingLoading
});

export const finishLoading = () => ({
    type: types.uiFinishLoading
});

export const startLogOut = () => {
    return async (dispatch) => {
        await firebase.auth().signOut();
        dispatch(logout());
        dispatch(notesLogout());
    }
} 

export const logout = () => ({
    type: types.logout
})