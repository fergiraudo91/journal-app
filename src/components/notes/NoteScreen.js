import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { activeNote, startDeleting } from '../../actions/notes';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar';


export const NoteScreen = () => {
    const dispatch = useDispatch();
    const {active} = useSelector(state => state.notes);
    const [values, handleInputChange, reset] = useForm(active);
    const activeID = useRef(active.id);

    const handleDelete = () => {

        dispatch(startDeleting(values.id));
    }

    useEffect(() => {
        if(active.id !== activeID.current){
            reset(active);
            activeID.current = active.id;
        }
        
    }, [active, reset]);

    useEffect(() => {
        
        dispatch(activeNote(values.id, {...values}));

    }, [values, dispatch]);

    return (
        <div className='notes__main-content'>
            <NotesAppBar date={active.date}/>
            <div className="notes__content animate__animated animate__fadeIn">

                <input
                    type="text"
                    placeholder="some awesome title"
                    className="notes__title-input"
                    autoComplete="off"
                    value={values.title}
                    onChange={handleInputChange}
                    name="title"
                />
                <textarea
                    className="notes__text-area"
                    placeholder="What happened today?"
                    value={values.body}
                    onChange={handleInputChange}
                    name="body"
                >

                </textarea>
                {
                    active.imgURL &&
                    <div className="notes__image">
                    <img src={active.imgURL}
                        alt="mountain" />

                </div>}

            </div>
            <button className="btn btn-danger"
            onClick={handleDelete}>Delete</button>
        </div>
    )
}
