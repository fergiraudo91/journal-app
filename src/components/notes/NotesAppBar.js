import moment from "moment";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startSaveNotes, startUploading } from "../../actions/notes";

export const NotesAppBar = ({ date }) => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);

  const dateTime = moment(date);

  const handleSave = () => {
    dispatch(startSaveNotes(note));
  };

  const handlePictureUpload = () => {
    document.getElementById('fileSelector').click();
  };

  const handleFileChange = (e) => {
    const file = e.target.files[0];
    console.log(file);
    if(file){
      dispatch(
        startUploading(file)
      );
    }


  };

  return (
    <div className="notes__appbar">
      <span>{`${dateTime.format("MMMM")} ${dateTime.format(
        "DD"
      )}, ${dateTime.format("YYYY")}`}</span>
      <input
        type="file"
        style={{ display: "none" }}
        onChange={handleFileChange}
        id="fileSelector"
        name="file"
      />
      <div>
        <button className="btn" onClick={handlePictureUpload}>
          Picture
        </button>
        <button className="btn" onClick={handleSave}>
          Save
        </button>
      </div>
    </div>
  );
};
