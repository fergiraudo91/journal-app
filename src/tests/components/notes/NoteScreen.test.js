import React from 'react';
import thunk from "redux-thunk";
import configureStore from 'redux-mock-store' //ES6 modules
import { mount } from "enzyme";
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import { NoteScreen } from "../../../components/notes/NoteScreen";
import { activeNote } from '../../../actions/notes';

jest.mock('../../../actions/notes', () => {
    return {
        activeNote: jest.fn()
    }
});


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {
    auth: {
        uid: 'dOjAFaDtmpQff5ByOQWfC4TUHcS2',
        name: 'Fernando Giraudo'
    },
    ui: {
        loading: false,
        msError: null
    },
    notes: {
        notes: [],
        active: {
            id: 'QI7eX0TxfyrUni6iNnCQ',
            body: 'Hola body',
            title: 'Hola titulo',
            date: 1625057652942,
            imgURL: 'https://res.cloudinary.com/dpezd0bbw/image/upload/v1625082320/qcjwk9p6kxddqlulykwe.png'
        }
    }
}

let store = mockStore(initState);
store.dispatch = jest.fn();

const wrapper = mount(
    <Provider store={store}>
        <NoteScreen />
    </Provider>
)

describe('Pruebas en <NoteScreen />', () => {


    test('Debe de mostrarse correctamente ', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de disparar el active note', () => {

        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'Hola Mundo'
            }
        });

        expect(activeNote).toHaveBeenCalledWith(
            'QI7eX0TxfyrUni6iNnCQ',
            {
                id: 'QI7eX0TxfyrUni6iNnCQ',
                body: 'Hola body',
                title: 'Hola Mundo',
                date: 1625057652942,
                imgURL: 'https://res.cloudinary.com/dpezd0bbw/image/upload/v1625082320/qcjwk9p6kxddqlulykwe.png'
            }
        )

    });


})
