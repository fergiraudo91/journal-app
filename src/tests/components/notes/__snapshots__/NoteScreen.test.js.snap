// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> Debe de mostrarse correctamente  1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      June 30, 2021
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content animate__animated animate__fadeIn"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="some awesome title"
      type="text"
      value="Hola titulo"
    />
    <textarea
      className="notes__text-area"
      name="body"
      onChange={[Function]}
      placeholder="What happened today?"
      value="Hola body"
    />
    <div
      className="notes__image"
    >
      <img
        alt="mountain"
        src="https://res.cloudinary.com/dpezd0bbw/image/upload/v1625082320/qcjwk9p6kxddqlulykwe.png"
      />
    </div>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
